math/src/field/fields/montgomery_backed_prime_fields.rs

```diff
@@ -197,6 +199,52 @@ where
     }
 }
 
+impl<C, const NUM_LIMBS: usize>
+    Distribution<FieldElement<MontgomeryBackendPrimeField<C, NUM_LIMBS>>> for Standard
+where
+    C: IsMontgomeryConfiguration<NUM_LIMBS> + Clone + Debug,
+{
+    fn sample<R: Rng + ?Sized>(
+        &self,
+        rng: &mut R,
+    ) -> FieldElement<MontgomeryBackendPrimeField<C, NUM_LIMBS>> {
+        let mut rand_limbs: [u64; NUM_LIMBS] = [0; NUM_LIMBS];
+        let mut is_msl: bool = true; //is Most Significative Limb of MODULUS
+
+        for i in 0..NUM_LIMBS {
+            match C::MODULUS.limbs[i] {
+                mod_limb if (mod_limb == 0) && is_msl => rand_limbs[i] = 0,
+                mod_limb if (i == (NUM_LIMBS - 1)) && is_msl => {
+                    rand_limbs[i] = rng.gen_range(1..mod_limb)
+                }
+                mod_limb if is_msl => {
+                    rand_limbs[i] = rng.gen_range(0..=mod_limb);
+                    is_msl = false;
+                }
+                _ if (i == 5) => {
+                    if rand_limbs == [0; NUM_LIMBS] {
+                        //to avoid 0 in the generation set
+                        rand_limbs[i] = rng.gen_range(1..=u64::MAX);
+                    } else {
+                        rand_limbs[i] = rng.gen_range(0..=u64::MAX);
+                    }
+                }
```

Comment:
This 5 seems a bit off. If it's NUM_LIMBS - 1 this case should never be reached, because is_msl is true and the first case matches.

Reply:
Sorry, my bad, the algo was originally designed for 6 limbs felts just fixed.